hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/CommonEntityAccessor.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.web3.state;
+
+import static com.hedera.services.utils.EntityIdUtils.entityIdFromId;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.mirror.common.domain.entity.Entity;
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.web3.repository.EntityRepository;
+import com.hedera.services.store.models.Id;
+import jakarta.annotation.Nonnull;
+import jakarta.inject.Named;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+
+@Named
+@RequiredArgsConstructor
+public class CommonEntityAccessor {
+    private final EntityRepository entityRepository;
+
+    public @Nonnull Optional<Entity> get(@Nonnull AccountID accountID, final Optional<Long> timestamp) {
+        if (accountID.hasAccountNum()) {
+            return getEntityByMirrorAddressAndTimestamp(
+                    entityIdFromId(new Id(accountID.shardNum(), accountID.realmNum(), accountID.accountNum())),
```

Comment:
We shouldn't use `Id` class as a intermediary. Instead we need need an `EntityIdUtils.toEntityId()` as I suggested in another PR.

```suggestion
                   toEntityId(accountID),
```